drivers/linstorvolumemanager.py

```diff
@@ -1132,7 +1120,13 @@ def update_volume_uuid(self, volume_uuid, new_volume_uuid, force=False):
                 'after volume uuid update: {}'.format(e)
             )
 
-        self._volumes.remove(volume_uuid)
+        try:
+            self._volumes.remove(volume_uuid)
+        except KeyError:
+            # Can be missing if we are building the volume set attr AND
+            # we are processing a deleted resource.
+            assert force
```

Comment:
```suggestion
        try:
            self._volumes.remove(volume_uuid)
        except KeyError as err:
            # Can be missing if we are building the volume set attr AND
            # we are processing a deleted resource.
            if not force:
                raise LinstorVolumeManagerError(
                    'Failed to remove old volume: {}'.format(err)
                )
```

Unless there is a very specific need for an assert, I'd prefer to have a clear exception message.

Reply:
In fact I don't see here a good reason to raise explicitely. First, I hesitated but there is already a verification of the volumes when the function is called without the force mode. So we can't have an exception with this design.

Reply:
So the assert is to catch a bug in a situation that should theoretically never happen?

Reply:
That's the definition of an assertion yes. :p

Reply:
fair point 😓